### DISCUSSION TITLE
Error when editing order

### DISCUSSION THREAD

**Discussion by jonandtice** [KmiQ/coinbase-advanced-python#56] [General]
Did something change? Using the 1.1 release (everything had been working fine) I'm suddenly getting the following error:
`  File "/home/jonandtice/.venv/coinbaseadvanced/lib/python3.11/site-packages/coinbaseadvanced/models/orders.py", line 472, in from_response
    raise CoinbaseAdvancedTradeAPIError.not_ok_response(response)
coinbaseadvanced.models.error.CoinbaseAdvancedTradeAPIError: {'error': 'PERMISSION_DENIED', 'error_details': 'User does not have access to portfolio', 'message': 'User does not have access to portfolio'}`

Is this happening to anyone else? Do I need to generate new API keys?

**Comment by KmiQ** [comment]
> Did something change? Using the 1.1 release (everything had been working fine) I'm suddenly getting the following error: ` File "/home/jonandtice/.venv/coinbaseadvanced/lib/python3.11/site-packages/coinbaseadvanced/models/orders.py", line 472, in from_response raise CoinbaseAdvancedTradeAPIError.not_ok_response(response) coinbaseadvanced.models.error.CoinbaseAdvancedTradeAPIError: {'error': 'PERMISSION_DENIED', 'error_details': 'User does not have access to portfolio', 'message': 'User does not have access to portfolio'}`
> 
> Is this happening to anyone else? Do I need to generate new API keys?

it sounds like an access issue, yes. Something I have noticed is that credentials need to match the portfolio where you are working. So, yeah, I'd advise you to try to play with your credentials and see if that fixes it. If your API key is for a portfolio != Default, I think you have to manually set your portfolio as part of the “retail_portfolio_id” arg in the payload (Coinbase constrain, no the library).

**Reply by jonandtice** [reply]
I only have the "Default" porfolio and everything had been working. This is a script that I run every day. I cancelled all of the open limit orders I was trying to edit, created new ones, and now I can edit those new orders. Must be something weird on Coinbase's end. I hope there isn't a limit to how many edits can be made to an order.

**Reply by jonandtice** [reply]
So this is still happening, every time I edit an order more than once. The second edit fails. I tried creating the orders with the Default retail_portfolio_id but that had the same effect. In fact, get_order returns kwargs={'retail_portfolio_id': ''} even when I manually specify the id when creating the order. I wold really prefer to avoid cancelling and creating orders when I should just be able to edit.

**Reply by KmiQ** [reply]
Yup, Delete and Creating sounds like a workaround. Is this happening to you as well via Coinbase UI? You probably want to report this issue to the Dev Forum (https://forums.coinbasecloud.dev/) so the Coinbase team is aware and fix it.
